app/services/migration/migrators/user.rb

```diff
@@ -54,6 +54,72 @@ def call
 
   private
 
+    def find_or_initialize_user(ecf_user)
+      # Find Users with `ecf_user.id`
+      users_with_ecf_id = ::User.where(ecf_id: ecf_user.id)
```

Comment:
Instead of loading the whole user model into memory here and in the queries below can we just pluck the `id` attributes and compare on that? has this been ran on the migration env yet?

Reply:
We'll need to return the User if found, no point running the query multiple times.